test/mainnet/withdrawer.spec.sol

```diff
@@ -0,0 +1,146 @@
+    }
+
+    function _convertAssetToBase(address asset_, uint256 assets) internal view returns (uint256) {
+        uint256 rate = provider.getRate(asset_);
+        return assets.mulDiv(rate, 10 ** 18, Math.Rounding.Floor);
+    }
+
+    function _convertBaseToAsset(address asset_, uint256 assets) internal view returns (uint256) {
+        uint256 rate = provider.getRate(asset_);
+        return assets.mulDiv(10 ** 18, rate, Math.Rounding.Floor);
+    }
+
+    function test_Vault_views() public {
+        assertEq(vault.asset(), MC.WETH, "Asset address should match");
+
+        uint256 totalAssets = INITIAL_BALANCE; // WBNB
+        totalAssets += _convertAssetToBase(MC.SLISBNB, INITIAL_BALANCE); // SLISBNB
+        vault.processAccounting();
+        assertEq(vault.totalAssets(), totalAssets, "Total assets should match");
+    }
+
+    function test_Vault_RequestWithdrawal_YNETH(uint256 amount) public {
+        vm.assume(amount > 1 ether);
+        vm.assume(amount < INITIAL_BALANCE / 2);
+
+        _requestWithdrawal(amount);
+    }
+
+    function test_Vault_ClaimWithdrawal_YNETH(uint256 amount) public {
+        vm.assume(amount > 1 ether);
+        vm.assume(amount < INITIAL_BALANCE / 2);
+
+        uint256 tokenId = _requestWithdrawal(amount);
+
+        _claimWithdrawal(tokenId);
+    }
+
+    function _requestWithdrawal(uint256 amount) internal returns (uint256 tokenId) {
+        address asset_ = MC.SLISBNB;
+        address stakeManager_ = MC.SLIS_BNB_STAKE_MANAGER;
+        ISlisBnbStakeManager stakeManager = ISlisBnbStakeManager(stakeManager_);
+
+        uint256 assets = vault.asyncWithdrawalBalance(asset_);
+        assertEq(assets, 0, "Queued assets should be zero");
+        vault.processAccounting();
+        uint256 totalAssets = vault.totalAssets();
+
+        processRequestWithdraw(vault, stakeManager_, asset_, amount);
+
+        ISlisBnbStakeManager.WithdrawalRequest[] memory requests =
+            stakeManager.getUserWithdrawalRequests(address(vault));
+        tokenId = requests.length - 1;
+
+        (bool _isClaimable, uint256 _amount) = stakeManager.getUserRequestStatus(address(vault), tokenId);
+        uint256 amountInBase = _convertAssetToBase(asset_, amount);
+
+        assertEq(_isClaimable, false, "Claimable should be false");
+        assertApproxEqRel(_amount, amountInBase, 1e15, "Amount should match");
+
+        assets = vault.asyncWithdrawalBalance(asset_);
+
+        assertApproxEqRel(assets, amountInBase, 1e15, "Queued assets should match");
+        assertApproxEqRel(vault.totalAssets(), totalAssets, 1e15, "Total assets should match");
+
+        vault.processAccounting();
+    }
+
+    function _claimWithdrawal(uint256 tokenId) internal {
+        address asset_ = MC.SLISBNB;
+        address stakeManager_ = MC.SLIS_BNB_STAKE_MANAGER;
+        ISlisBnbStakeManager stakeManager = ISlisBnbStakeManager(stakeManager_);
+        MockStakeHub stakeHub = MockStakeHub(MC.SLIS_BNB_STAKE_HUB);
```

Comment:
Can we test this without a mock?

it's important here we test against the real thing